common-rendering/src/keyEvent.ts

```diff
@@ -0,0 +1,25 @@
+import { palette } from "@guardian/src-foundations";
+import { Pillar, Theme } from "@guardian/types";
+
+export type paletteId = 300 | 400;
+
+export interface KeyEvent {
+	time: string;
+	text: string;
+	url: string;
+}
+
+export const getColor = (pillar: Theme, paletteId: paletteId) => {
```

Comment:
You could use [decidePalette](https://github.com/guardian/dotcom-rendering/blob/a42df26f3e33fd7340bdea059a3f67bf91b678db/dotcom-rendering/src/web/lib/decidePalette.ts) here. This file is used in DCR to decide colours and has proven to be super useful!

Reply:
Do we need to do a separate investigation on how to use palette colours between AR & DCR?

Reply:
Yeah, I think that would be a good idea.

I should point out that the reason `decidePalette` works so well is it is very common for the colour logic to change (design tweaks / new article types / etc.) and it's much easier to change things in one location - where you can see the scope of what you're working on very easily - vs. having to search the codebase for potential impacts/areas that need updating which is hard making it easy to miss things. This pattern coupled with visual regression gave us win after win after win.

Reply:
Oh, and another reason `decidePalette` is my favourite thing is because it makes it really easy for non engineers to edit colours. So a designer can spot a contrast problem, say the Kicker text for Labs articles looks too grey inside a Media Card, and they can read the code themselves and fix it. They can even write additional logic if needed. The regression tests then take these changes and make sure nothing else is being affected. This is so, so great.

Reply:
Great thanks @oliverlloyd , I had seen the decidePalette file and thought that'd be a good fit for here. But as Olly said, we can update the code after we decide on how to share the decidePalette between the AR & dotcom

Reply:
I'm not sure we should link the use of `decidePalette` with the decision on how to share it with AR. Even if we decide to not share it with AR, we should still continue to use it in DCR because this is the convention.

We could decide not to use this convention - that's certainly an option - but until then, keeping things consistent makes the most sense I think - just in terms of maintainability and ease of understanding it's always nice if there are consistent ways for doing things.

Reply:
Oh, I _just_ saw that this code is inside `common-rendering`. I _see_ now what you're talking about. Sorry, I missed that!

Hmm, maybe it would be a good idea to think about these problems - how to share code - sooner rather than later? It's great having a monorepo but if we don't implement methods for sharing code then we're only going to be co-locating code and not getting any of the actual benefits.

Reply:
Umm, I guess the solution here is simply to hoist the utils function `decidePalette` up into a shared folder in the root. Maybe `/shared/decidePalette.ts` or `/shared/utils/decidePalette.ts`?

You _could_ make this folder an npm package but I don't think you need to right away, not unless we decide we need versioning. But this is something that be added later on as required.

I'm happy to have a bash at this if no-one has any objections? It feeds into some work I've got coming up

Reply:
Thanks Oliver for giving great detailed info. We had a chat with everyone and the team is still unsure on the colour logic. Right now AR and dotcom are having different approaches on how to get the colour and we're trying to find what'd be the best for this common-rendering project that'd be used by both AR & DCR. 

So for now we decided to leave the colour logic within the components until we come up with a utility (either using decidePalette or a new util file) to get the colours. I think @OllysCoding is going to chat with you about this too :)

Reply:
Hey @marjisound sounds good 👍 and sorry for hijacking your PR!

Reply:
Not at all, I'd appreciate your time and your feedback :)